solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/RankSVMModel.java

```diff
@@ -28,36 +29,62 @@
 
 public class RankSVMModel extends LTRScoringModel {
 
-  protected float[] featureToWeight;
+  protected Float[] featureToWeight;
 
   /** name of the attribute containing the weight of the SVM model **/
   public static final String WEIGHTS_PARAM = "weights";
 
-  public RankSVMModel(String name, List<Feature> features,
+  public static RankSVMModel create(String name, List<Feature> features,
```

Comment:
We should mimic what you did for the Features class here:
https://github.com/bloomberg/lucene-solr/blob/master-ltr-plugin-develop/solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/Feature.java#L47

Move this to a generic getInstance() function in the parent class.